spark/src/main/scala/org/apache/mahout/sparkbindings/blas/ABt.scala

```diff
@@ -116,7 +118,12 @@ object ABt {
           // Empty combiner += value
           createCombiner = (t: (Array[K], Array[Int], Matrix)) =>  {
             val (rowKeys, colKeys, block) = t
-            val comb = new SparseMatrix(prodNCol, block.nrow).t
```

Comment:
it may do some good (do you have any experimental evidence, btw?), but keep in mind that "SPARSELIKE" is indication of the backing structure being used, not the actual data sparsity. 

I think elsewhere i hacked this (or at least i remember trying to hack it this way) by creating quick random sparsity analysis (select few random elements to estimate sparsity) and then use proper type of matrix if you end up on either side of some threshold (say, 25% non-sparse). As you keep summing out the blocks, even if they start sparse, they are quite likely to turn dense (in data sense) down the rank update chain. Products of sparse arguments usually tend to be densier (e.g., what if in C=AB, just 1 column in A and just 1 row in B happen to be 100% dense, while otherwise being extremely sparse arguments?).